bokehjs/src/coffee/models/plots/plot_canvas.coffee

```diff
@@ -338,6 +338,22 @@ class PlotCanvasView extends Renderer.View
             if is_panning?
               range_info['start'] = rng.get('start')
 
+    min_interval = rng.get('min_interval')
+    max_interval = rng.get('max_interval')
+    if min_interval? || max_interval?
+      new_interval = Math.abs(range_info['end'] - range_info['start'])
+      center = 0.5 * (rng.get('end') + rng.get('start'))
+      sign = if reversed then -1 else 1
+
+      if min_interval > 0
+        if new_interval < min_interval
+          range_info['start'] = center - sign * 0.5 * min_interval
+          range_info['end'] = center + sign * 0.5 * min_interval
+      if max_interval > 0
+        if new_interval > max_interval
+          range_info['start'] = center - sign * 0.5 * max_interval
+          range_info['end'] = center + sign * 0.5 * max_interval
```

Comment:
> I would personally refuse to update the range if either range constraint would be violated.

I would suggest starting with constraining individually as that's always been our behavior to date. I think it would be nice if we could offer what you're suggesting as a convenience wrapper / extra mode.  I can see how why what your suggesting would be useful in most cases, but I think it's a really low-level place to enforce that and it might not suit some people. We already have where you can be over an axis and zoom independently. This feels like it might clash with that.